src/v/pandaproxy/schema_registry/service.cc

```diff
@@ -55,6 +54,15 @@ const security::acl_principal principal{
 
 class wrap {
 public:
+    enum class auth_level {
+        // Unauthenticated endpoint (not a typo, 'public' is a keyword)
+        publik = 0,
+        // Requires authentication (if enabled) but not superuser status
+        user = 1,
+        // Requires authentication (if enabled) and superuser status
+        superuser = 2
+    };
```

Comment:
if the enumerator values (e.g. 0,1,2) aren't important, it's better to remove them (e.g. their values neither enter nor escape redpanda process). if they are important, a note about where their values interact with the outside world is a useful comment to have.